yt/visualization/volume_rendering/scene.py

```diff
@@ -437,15 +443,55 @@ def save_annotated(
         """
         fname = self._setup_save(fname, render)
 
-        # which transfer function?
-        rs = self._get_render_sources()[0]
-        tf = rs.transfer_function
-        label = rs.data_source.ds._get_field_info(rs.field).get_label()
-
         ax = self._show_mpl(
             self._last_render.swapaxes(0, 1), sigma_clip=sigma_clip, dpi=dpi
         )
-        self._annotate(ax.axes, tf, rs, label=label, label_fmt=label_fmt)
+
+        # number of transfer functions?
+        num_trans_func = 0
+        for rs in self._get_render_sources():
+            try:
+                tf = rs.transfer_function
+                num_trans_func += 1
+            except AttributeError:
+                print("error")
+                pass
```

Comment:
Why should this error be caught ? I wonder wether there's a legit reason (in which case the `except` block should contain more information), or if we should simply drop the `try/except` struct.

Reply:
it's been so long since I initially wrote this -- but I think some sources might not have transfer functions (like a point source), so we want to skip them.

Reply:
okay, indeed, I looked at the original vis I did with this, and I had 2 volume sources and 1 point source, so we need to skip the point source, since that has no transfer function.

Reply:
Alright thanks for the explanation. In that case here's a refactor suggestion
```suggestion
            if hasattr(rs, "transfer_function"):
                num_trans_func += 1
```

Reply:
I like that.  Thanks.